src/redux/throttleMiddleware.ts

```diff
@@ -0,0 +1,48 @@
+import { VALUE_CHANGED, Action, VALUES_CHANGED } from "./actions";
+
+export class UpdateThrottle {
+  private queue: Action[];
+  private started: boolean;
+  private updateMillis: number;
+  public ready: boolean;
+  constructor(updateMillis: number) {
+    this.queue = [];
+    this.started = false;
+    this.updateMillis = updateMillis;
+    this.ready = true;
+    this.queueUpdate = this.queueUpdate.bind(this);
+    this.clearQueue = this.clearQueue.bind(this);
+    this.setReady = this.setReady.bind(this);
+  }
+
+  public queueUpdate(action: Action): void {
+    this.queue.push(action);
+    if (!this.started) {
+      setInterval(this.setReady, this.updateMillis);
+      this.started = true;
+    }
+  }
+
+  public setReady(): void {
+    this.ready = true;
+  }
+
+  public clearQueue(store: any): void {
+    store.dispatch({ type: VALUES_CHANGED, payload: [...this.queue] });
+    this.queue.length = 0;
+    this.ready = false;
+  }
+}
+
+export const throttleMiddleware = (updater: UpdateThrottle) => (store: any) => (
+  next: any
+) => (action: any): any => {
+  if (action.type === VALUE_CHANGED) {
```

Comment:
Should this also be able to handle `VALUES_CHANGED`? Perhaps in the future the server or complex widgets could provide bundled value updates which would want to use this mechanism?

Reply:
Yes, that sounds very possible. I will leave it as it is, because if we do want such a feature it will be possible to add it at the time and it's best to do it when you know more clearly what you need.

Reply:
Ok, I guess in terms of complex widgets we would need to write a custom dispatcher and therefore a new hook anyway?